lib/pg_search/features/tsearch.rb

```diff
@@ -28,8 +24,47 @@ def rank
         arel_wrap(tsearch_rank)
       end
 
+      def highlight
+        arel_wrap(ts_headline)
+      end
+
       private
 
+      def checks_for_prefix
+        if options[:prefix] && model.connection.send(:postgresql_version) < 80400
+          raise PgSearch::NotSupportedForPostgresqlVersion.new(<<-MESSAGE.strip_heredoc)
+            Sorry, {:using => {:tsearch => {:prefix => true}}} only works in PostgreSQL 8.4 and above.")
+          MESSAGE
+        end
+      end
+
+      def checks_for_highlight
+        if options[:highlight] && model.connection.send(:postgresql_version) < 90000
+          raise PgSearch::NotSupportedForPostgresqlVersion.new(<<-MESSAGE.strip_heredoc)
+            Sorry, {:using => {:tsearch => {:highlight => true}}} only works in PostgreSQL 9.0 and above.")
+          MESSAGE
+        end
+      end
+
+      def ts_headline
+        "ts_headline((#{document}), (#{tsquery}), '#{ts_headline_options}')"
+      end
+
+      def ts_headline_options
+        return nil unless options[:highlight].is_a?(Hash)
+
+        headline_options = map_headline_options
+        headline_options.map{|key, value| "#{key} = #{value}" if value }.compact.join(", ")
+      end
+
+      def map_headline_options
+        {
+          "StartSel" => options[:highlight][:start_sel],
+          "StopSel" => options[:highlight][:stop_sel],
+          "MaxFragments" => options[:highlight][:max_fragments]
+        }
```

Comment:
What do you think of allowing other options? 

From [the docs](http://www.postgresql.org/docs/current/static/textsearch-controls.html#TEXTSEARCH-HEADLINE):

- FragmentDelimiter
- HighlightAll
- ShortWord
- MaxWords, MinWords